File: quiz_hub/models/card_option.py
from typing import TYPE_CHECKING

from sqlalchemy import String, Text, DateTime, ForeignKey, Float
from sqlalchemy.orm import Mapped, mapped_column, relationship
from database import Base
from datetime import datetime, timezone

if TYPE_CHECKING:
    from .card import Card





class Card_option(Base):
    __tablename__ = "card_options"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True, index=True)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    is_right: Mapped[bool] = mapped_column(nullable=False)
    card_id: Mapped[int] = mapped_column(ForeignKey("cards.id"), nullable=False)

    card: Mapped["Card"] = relationship("Card", uselist=False, back_populates="options")

File: quiz_core/internal/models/card_option.go
package models

type CardOption struct {
	ID      uint   `gorm:"column:id;primaryKey;autoIncrement;index:ix_card_options_id" json:"id"`
	Content string `gorm:"column:content;type:text;not null" json:"content"`
	IsRight bool   `gorm:"column:is_right;not null" json:"is_right"`
	CardID  uint   `gorm:"column:card_id;not null" json:"card_id"`

	Card Card `gorm:"foreignKey:CardID;references:ID;constraint:OnDelete:CASCADE" json:"card,omitempty"`
}

func (CardOption) TableName() string {
	return "card_options"
}
